language/expressions.md

```diff
@@ -720,9 +726,26 @@ result. Chained assignments are permitted.
 
 Examples:
 
+~~~puppet
+
     $a = 10
     $x = $y = 0
 
+    # from array
+    [$a, $b] = [1, 2]  # assigns 1 to $a, and 2 to $b
+
+    # from hash
+    [$a, $b] = { a => 10, b => 20, c => 30 }  # assigns 10 to $a, and 20 to $b
+
+    # from class
+    class mymodule::someclass::example($x = 100) {
+      $a = 10
+    }
+    include example
+    [$a, $x] = Class['mymodule::someclass::example']  # assigns 10 to $a, and 100 to $x
```

Comment:
You cannot extract from a define, only from classes. No plan to add it to defines because they are evaluation order dependant, and it would only be their parameters that can be accessed. Also, the final values of resource parameters are in some cases lazily determined (because of reource defaults expressions, and overrides). So - no, too much whimsical old behaviour involved there.